sim/firesim-lib/src/main/scala/configs/CompilerConfigs.scala

```diff
@@ -83,7 +83,6 @@ class WithAutoCounterPrintf extends Config((site, here, up) => {
 class BaseF1Config extends Config(
   new WithWiringTransform ++
   new WithAsyncResetReplacement ++
-  new WithPlusArgReaderRemoval ++
```

Comment:
@timsnyder-siv Remind me why this had to be removed again?

Reply:
From the commit message when I took it out ff837e5a0dd0b5b34204905b720ca9755c5f23b3:

> The PlusArgReaderPass changes extmodule plusarg_reader to
a module that drives its output from DEFAULT but does not remove
BlackBox* Annotations.  It relies on DeadCodeElimination coming along
later to clean up the [dead blackboxes]
> 
> Rather than using the PlusArgReaderPass, the verilog SYNTHESIS `define
accomplishes the same effect in the plusarg_reader blackbox verilog.

Now that we have DCE re-enabled it might be possible to put this back in.  Want me to check?

My question would be why put it back in?  The PlusArgReader's don't seem to cause problems for downstream transforms (or there are silent problems that need to be made noisier).  They are also useful during MIDAS-level simulation.

Reply:
Yeah it seems to me that everything works fine now with it out. We put it in initially because MIDAS couldn't process black boxes. If the PlusArgReader interacts sanely with ML sim I see little reason to remove it.